exec/java-exec/src/main/java/org/apache/drill/exec/store/sys/store/provider/LocalPersistentStoreProvider.java

```diff
@@ -57,14 +59,14 @@ public LocalPersistentStoreProvider(final DrillConfig config) throws StoreExcept
   }
 
   @Override
-  public <V> PersistentStore<V> getOrCreateStore(PersistentStoreConfig<V> storeConfig) {
+  public <V> PersistentStore<V> getOrCreateStore(PersistentStoreConfig<V> storeConfig) throws StoreException {
     switch(storeConfig.getMode()){
     case BLOB_PERSISTENT:
     case PERSISTENT:
-      if (enableWrite) {
+      if (!useEphemeral) {
```

Comment:
I've actually refactored the internal unit tests that leveraged this flag. It doesn't seem to be used outside of the tests.
I think the enableWrite is a little misleading. We already have a skipWritingProfiles flag introduced. Do you really think there is a backward compatibility that would be broken, considering this was hidden away in a 'test' package?